### PR TITLE
Hotfix/workflow friendly

### PR DESCRIPTION
**Pull request by stuartromanek**

### CODE REVIEW

File: lib/api.js [hunk 1]
```diff
@@ -17,22 +28,43 @@ module.exports = (self, options) => {
...
     }
   };
 
   self.findOne = async req => {
     try {
-      const dialog = await self.apos.modules['apostrophe-dialog-box']
-        .find(req, { _id: req.params.id })
-        .toObject();
+      let dialog;
+      if (req.session.workflowMode && req.session.workflowMode === 'live') {
```
Comment (boutell) [inline lib/api.js]: I don't think this will be set for anonymous site visitors.

File: lib/api.js [hunk 2]
```diff
@@ -17,22 +28,43 @@ module.exports = (self, options) => {
...
+        dialog = await self.apos.modules['apostrophe-dialog-box']
+          .find(req, { _id: req.params.id })
+          .toObject();
 
-      return dialog;
-    } catch (error) {
-      return null;
+        if (!dialog) {
+          // at this point we likely need to go look for the live version
+          const get = Promise.promisify(self.getDraftAndLiveWrapper);
```
Comment (boutell) [inline lib/api.js]: How did this problem get started? How is the id of the dialog box being stored? If it is a joinByOne then committing should be translating it to the id of the live version of the same dialog.

**Comment by stuartromanek** [comment]
YEah, this does not work for anon users, okay...

This comes up because of the way the custom CKEditor implementation of inserting a link that that triggers a dialog box works.. It stores the piece id as part of the rich text, so its technically not a join and doesn't enjoy whatever magic figures which document to grab.

What is the right entry point for solving this? Should I try to be converting the id at rich text render or have some special rich text rendering route that does something different altogether?

**Comment by boutell** [comment]
You can hook into the mechanism apostrophe-workflow uses to resolve relationships across locales by adding an event handler:

https://github.com/apostrophecms/apostrophe-rich-text-permalinks/blob/master/lib/modules/apostrophe-rich-text-permalinks-rich-text-widgets/index.js#L125-L178

**Comment by stuartromanek** [comment]
⚡️ ⚡️ ⚡️

File: lib/routes.js [hunk 3]
```diff
@@ -1,17 +1,20 @@
...
 
-      if (box) {
-        return res.status(200).send(self.render(
-          req,
-          `apostrophe-dialog-box:dialogs/${box.template}.html`,
-          { piece: box }
-        ));
+        return res.status(404).send('');
+      } catch (error) {
+        console.log(error);
```
Comment (boutell) [inline lib/routes.js]: You already caught the error in findOne so you won't get here. Recommend you remove the catch in findOne and use this one catch to do the right thing for all of the error cases.

Comment (stuartromanek) [reply lib/routes.js]: 👁

**Pull request merged**